### PR TITLE
feat: PDM support, ruff using pre-commit, typesafe code

### LINKED ISSUE #5: [Feature] Add Support for PDM
### Describe the feature you'd like to request

PDM Support

### Describe the solution you'd like

PDM Support

### Describe alternatives you've considered

_No response_

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by SAMAD101**
# Description

Please include a summary of the change and which issue is fixed. Please also include relevant motivation and context. List any dependencies that are required for this change.

Fixes #5, #4 and #11

## Type of change
Just added PDM support
Please delete options that are not relevant.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [x] This change requires a documentation update

# How Has This Been Tested?

Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration

- [ ] Test A
- [ ] Test B

**Test Configuration**:
* Firmware version:
* Hardware:
* Toolchain:
* SDK:

# Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [ ] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

File: src/bitssh/prompt.py [hunk 1]
```diff
@@ -7,7 +7,7 @@
 
 class Prompt:
     @classmethod
-    def ask_host_prompt(cls):
+    def ask_host_prompt(cls) -> None:
         HOST = ConfigPathUtility.get_config_file_host_data()
```
Comment (Mr-Sunglasses) [inline src/bitssh/prompt.py:11]: This variable should be typed

File: src/bitssh/prompt.py [hunk 2]
```diff
@@ -7,7 +7,7 @@
...
         questions = [
```
Comment (Mr-Sunglasses) [inline src/bitssh/prompt.py:12]: This should also be typed

**Review by Mr-Sunglasses — COMMENTED** [review]
Also to ensure that all the files is typed we need to add `py.typed` empty file to `src/bitssh/`

File: pyproject.toml [hunk 3]
```diff
@@ -1,14 +1,16 @@
 [build-system]
-requires      = ["setuptools>=61.0.0", "wheel"]
-build-backend = "setuptools.build_meta"
+requires = ["pdm-backend"]
+build-backend = "pdm.backend"
```
[the remark this answers is not in the document]
Comment (SAMAD101) [reply pyproject.toml:3]: understood

File: pyproject.toml [hunk 4]
```diff
@@ -18,6 +20,8 @@ keywords = ["bitssh", "sshmanager", "commandline"]
 dependencies = [
```
[the remark this answers is not in the document]
Comment (SAMAD101) [reply pyproject.toml:20]: sure

**Review by Mr-Sunglasses — CHANGES_REQUESTED** [review]
Any updates?

**Comment by SAMAD101** [comment]
yeah, did a little refactor and added `py.typed` file in `src/bitssh/`

**Comment by Mr-Sunglasses** [comment]
@SAMAD101 Can you please resolve the conflicts.

**Comment by SAMAD101** [comment]
> @SAMAD101 Can you please resolve the conflicts.

sure
seems like my branch was too old, that's what caused the conflists

**Review by Mr-Sunglasses — APPROVED** [review]
Thanks @SAMAD101

**Pull request merged**